element_calcium_imaging/export/nwb/nwb.py

```diff
@@ -0,0 +1,339 @@
+        )
+        _add_segmentation_data_to_nwb(
+            session_key,
+            nwbfile=nwb_file,
+            imaging_plane=nwb_file.imaging_planes["ImagingPlane"],
+        )
+
+    else:
+        imaging_plane = _add_scan_to_nwb(session_key, nwbfile=nwb_file)
+        _add_image_series_to_nwb(session_key, imaging_plane=imaging_plane)
+        _add_segmentation_data_to_nwb(
+            session_key, nwbfile=nwb_file, imaging_plane=imaging_plane
+        )
+
+    return nwb_file
+
+
+def _create_raw_data_nwbfile(session_key, linked_nwb_file):
+    """Adds raw data to NWB file.
+
+    Args:
+        session_key (dict): key from Session table
+        linked_nwb_file (NWBFile): nwb file
+    """
+
+    acquisition_software = (scan.Scan & session_key).fetch1("acq_software")
+    frame_rate = (scan.ScanInfo & session_key).fetch1("fps")
+
+    if acquisition_software == "NIS":
+        raise NotImplementedError(
+            "Packaging raw data acquired from `Nikon NIS Elements` software is not supported at this time."
+        )
+
+    elif acquisition_software == "PrairieView":
+        n_planes = (scan.ScanInfo & session_key).fetch1("ndepths")
+        raw_data_files_location = get_calcium_imaging_files(
+            session_key, acquisition_software
+        )
+
+        if n_planes > 1:
+            from neuroconv.converters import (
+                BrukerTiffMultiPlaneConverter as BrukerTiffConverter,
+            )
+
+            imaging_interface = BrukerTiffConverter(
+                file_path=raw_data_files_location[0],
+                fallback_sampling_frequency=frame_rate,
+                plane_separation_type="disjoint",
+            )
+        else:
+            from neuroconv.converters import (
+                BrukerTiffSinglePlaneConverter as BrukerTiffConverter,
+            )
+
+            imaging_interface = BrukerTiffConverter(
+                file_path=raw_data_files_location[0],
+                fallback_sampling_frequency=frame_rate,
+            )
+        metadata = imaging_interface.get_metadata()
+        imaging_interface.run_conversion(
+            nwbfile=linked_nwb_file,
+            metadata=metadata,
+        )
+    else:
+        if acquisition_software == "ScanImage":
+            from neuroconv.datainterfaces import (
+                ScanImageImagingInterface as ImagingInterface,
+            )
+        elif acquisition_software == "Scanbox":
+            from neuroconv.datainterfaces import SbxImagingInterface as ImagingInterface
+
+        raw_data_files_location = get_calcium_imaging_files(
+            session_key, acquisition_software
+        )
+
+        imaging_interface = ImagingInterface(
+            file_path=raw_data_files_location[0], fallback_sampling_frequency=frame_rate
+        )
+        metadata = imaging_interface.get_metadata()
+        imaging_interface.add_to_nwbfile(
+            nwbfile=linked_nwb_file,
+            metadata=metadata,
+        )
```

Comment:
You didn't have to do something like `imaging_interface.run_conversion()` here? I don't know myself, just curious as that was the pattern above

Reply:
Yes, that had to be changed because the `run_conversion()` method saves an NWB file whereas `add_to_nwbfile()` is better-suited for our application where we will allow the user to pick where they want to save it.